Source/JavaScriptCore/bytecode/UnlinkedCodeBlock.cpp

```diff
@@ -345,4 +345,50 @@ void UnlinkedCodeBlock::allocateSharedProfiles(unsigned numBinaryArithProfiles,
     m_unaryArithProfiles = FixedVector<UnaryArithProfile>(numUnaryArithProfiles);
 }
 
+RefPtr<MetadataTable> UnlinkedCodeBlock::metadataLink()
+{
+    if (!m_metadata->isLinked() && m_cachedMetadataIndexes) {
+        // TODO: What if m_metadata is not linked but m_cachedMetadataIndexes is valid.
+        // That means there is a case that
+        //   1. One metadata table was linked
+        //   2. Cache happened.
+        //   3. The metadata table was unlinked. But the cache info stored in the  is not cleared
+        // So, I am thinking to copy the cached info locally first. Then, do `link` and copy the cached info
+        // back to the metadata table:
+        //
+        // cacheInfo = m_metadata->asMetadataTable()->cacheInfo(m_cachedMetadataIndexes);
+        // metadataTable = m_metadata->link();
+        // metadataTable->embed(cacheInfo);
+        // return metadataTable;
```

Comment:
Ok, I think we can do similar thing as
```
struct DeallocTable {
    template<typename Op>
    static void withOpcodeType(MetadataTable* table)
    {
        if constexpr (static_cast<unsigned>(Op::opcodeID) < NUMBER_OF_BYTECODE_WITH_METADATA) {
            table->forEach<Op>([](auto& entry) {
                entry.~Metadata();
            });
        }
    }
};
```
when we clear the first metadata table but keeping the cache info.